src/mame/tandberg/tdv2100_disp_logic.h

```diff
@@ -0,0 +1,141 @@
+	auto write_ackl_callback() { return m_write_ackl_cb.bind(); }
+	auto write_nakl_callback() { return m_write_nakl_cb.bind(); }
+
+	DECLARE_INPUT_CHANGED_MEMBER(uart_changed);
+
+	// Input strobes
+	void process_keyboard_char(uint8_t key);
+	void cleark_w(int state);
+	void linek_w(int state);
+	void transk_w(int state);
+	void break_w(int state);
+
+protected:
+	virtual void device_add_mconfig(machine_config &config) override;
+	virtual void device_start() override;
+	virtual void device_reset() override;
+	virtual ioport_constructor device_input_ports() const override;
+	virtual const tiny_rom_entry *device_rom_region() const override;
+
+private:
+
+	uint32_t screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect);
+
+	// Video, Cursor & Sound
+	void vblank(int state);
+	int get_attribute(int at_row, int at_col, int attribute = 0, int from_row = 0, int from_col = 0);
+	void char_to_display(uint8_t byte);
+	void data_to_display(uint8_t byte);
+	void advance_cursor();
+	void place_cursor(int row, int col);
+	void clear_key_handler();
+	int get_ram_addr(int row, int col);
+	void erase_row(int row);
+	TIMER_CALLBACK_MEMBER(expire_speed_check);
+	TIMER_CALLBACK_MEMBER(end_beep);
+
+	// RS-232
+	void rs232_rxd_w(int state);
+	void rs232_dcd_w(int state);
+	void rs232_dsr_w(int state);
+	void rs232_cts_w(int state);
+	void rs232_txd_w(int state);
+	void rs232_ri_w(int state);
+	void uart_rx(int state);
+	void uart_tx(int state);
+	void update_rs232_lamps();
+	void check_rs232_rx_error(int state);
+	void set_uart_state_from_switches();
+	void clock_on_line_flip_flop();
+	void clock_transmit_flip_flop();
+
+	required_device<screen_device>      m_screen;
+	required_device<palette_device>     m_palette;
+	required_region_ptr<uint8_t>        m_font;
+	required_region_ptr<uint8_t>        m_addr_offsets;
+	memory_share_creator<uint8_t>       m_vram;
+	required_device<beep_device>        m_beep;
+	required_device<ay51013_device>     m_uart;
+	required_device<clock_device>       m_uart_clock;
+	required_device<rs232_port_device>  m_rs232;
+	required_ioport                     m_sw_invert_video;
+	required_ioport                     m_sw_rs232_baud;
+	required_ioport                     m_sw_rs232_settings;
+	required_ioport                     m_dsw_u39;
+	required_ioport                     m_dsw_u73;
+	required_ioport                     m_dsw_u61;
+
+	devcb_write_line                    m_write_waitl_cb;
+	devcb_write_line                    m_write_onlil_cb;
+	devcb_write_line                    m_write_carl_cb;
+	devcb_write_line                    m_write_errorl_cb;
+	devcb_write_line                    m_write_enql_cb;
+	devcb_write_line                    m_write_ackl_cb;
+	devcb_write_line                    m_write_nakl_cb;
+
+	// Video, Cursor & Sound
+	int m_frame_counter;
+	bool m_video_enable;
+	bool m_vblank_state;
+	int m_cursor_row;
+	int m_cursor_col;
+	int m_page_roll;
```

Comment:
Please use explicitly sized integer types for data members that need to be registered for save states for portability.

Reply:
Fixed